### ISSUE TITLE
OpenID redirect issue

### ISSUE THREAD

**Issue by patvdleer** [pennersr/django-allauth#1314]
Everything works in my local dev but as soon as I run the service in production the openid login fails.

in dev the following happens:
* http://127.0.0.1:8000/en/openid/login/?process=login
* I enter http://steamcommunity.com/openid
* redirects me to steam using -> `https://steamcommunity.com/openid/login?openid.ax.mode=fetch_request&openid.ax.required=ext0%2Cext1%2Cex...`
* redirects me back to either login or social account create

On production however the http://127.0.0.1:8000/en/openid/login/?process=login redirects me to localhost with the parameters that should go to steam instead of localhost, like `http://127.0.0.1:8000/en/openid/login/?openid.ax.mode=fetch_request&openid.ax.required=ext0%2Cext1%2Cext2%2Cext3`

~~~
ERROR:root:Missing required parameter in response from https://steamcommunity.com/openid/login: ('http://specs.openid.net/auth/2.0', 'assoc_type')
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/openid/message.py", line 455, in getArg
    return self.args[args_key]
KeyError: ('http://specs.openid.net/auth/2.0', 'assoc_type')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/openid/consumer/consumer.py", line 1280, in _requestAssociation
    assoc = self._extractAssociation(response, assoc_session)
  File "/usr/local/lib/python3.4/dist-packages/openid/consumer/consumer.py", line 1397, in _extractAssociation
    OPENID_NS, 'assoc_type', no_default)
  File "/usr/local/lib/python3.4/dist-packages/openid/message.py", line 458, in getArg
    raise KeyError((namespace, key))
KeyError: ('http://specs.openid.net/auth/2.0', 'assoc_type')
~~~

**Comment by patvdleer** [comment]
For more info please see http://stackoverflow.com/questions/35695666/django-allauth-openid-redirect-apache-as-proxy

**Comment by patvdleer** [comment]
http://stackoverflow.com/questions/35695666/django-allauth-openid-and-oauth-redirect-apache-as-proxy/35699034#35699034

**Comment by therefromhere** [comment]
We hit exactly the same error message, with an extra error before like this "invalid request block size: 4167 (max 4096)...skip" 

This occurred because uwsgi was truncating the openid url.

The fix is to increase the uwsgi buffer size:

```
[uwsgi]
# allow long urls (default is 4096)
buffer-size=32768
...
```

**Comment by therefromhere** [comment]
Having investigated this further, I suspect everyone using allauth for steam login is actually hitting this error currently, though they might not be seeing it in their logs depending on logging config, since openid logs directly to the root logger. 

I think this would be resolved by #1814 (and adding stateless=True to the SERVERS config as per the PR).

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1314
